### PR TITLE
Surfs up

### PR DESCRIPTION
**Pull request by RobotCaleb**
image

### CODE REVIEW

**Comment by RobotCaleb** [comment]
closes #418

**Comment by RobotCaleb** [comment]
with this, mesher data can lose material index, no?

**Comment by chrisforbes** [comment]
Possibly -- we may still need it to select different palettes.

**Comment by chrisforbes** [comment]
(Palette texture changes from 1D to 1DArray)

File: src/tools/remove_surface.cc [hunk 1]
```diff
@@ -52,8 +52,10 @@ struct remove_surface_tool : tool
         if (!can_use(rc))
             return;
 
+        block *bl = rc->block;
         auto index = normal_to_surface_index(rc);
-        auto mesh = asset_man.get_surface_mesh(index);
+
+        auto mesh = asset_man.get_surface_mesh(index, bl->surfs[index]);
```
Comment (chrisforbes) [inline src/tools/remove_surface.cc]: What's the path forward here for variants? We pass the location to get_surface_mesh as well?

**Pull request merged**